src/modules/rlm_sql/rlm_sql.h

```diff
@@ -170,6 +171,25 @@ typedef struct {
 
 typedef struct sql_grouplist rlm_sql_grouplist_t;
 typedef struct sql_thread_group rlm_sql_thread_group_t;
+
+/** Context data for rlm_sql_select_query
+ *
+ */
+typedef enum {
+	SELECT_QUERY_START = 0,
+	SELECT_QUERY_RESUME,
+} sql_thread_select_query_state_t;
```

Comment:
Again functions are better than enums, and it's more consistent with the two_step function you added.